### ISSUE TITLE
createCollection improvements

### ISSUE THREAD

**Issue by SachaG** [VulcanJS/Vulcan#2213]
I think we could make a couple improvements to `createCollection`. For example it could be used to: 

- add custom resolvers and mutations
- define views
- define access restrictions

Basically, centralize everything that's currently defined with `Collection.doSomething()` in one place to make the APIs cleaner. Any other ideas?

**Comment by SachaG** [comment]
Although one thing to take into account is the client/server code split. Things like resolvers and mutations may only make sense on the server, while `createCollection` is currently supposed to run on both…

**Comment by SachaG** [comment]
Maybe something like this would work:

```

// client + server

export const Posts = createCollection({

  collectionName: 'Posts',

  typeName: 'Post',

  schema,
  
  checks: {
    read: {},
    create: {},
    update: {},
    delete: {},
  }

  views: {
    default: ...,
    top: ...,
    new: ...,
  }

});

// server only

import Posts from '../modules/posts/collection.js';

extendCollection(Posts, {

  resolvers: [
    ...getDefaultResolvers('Posts'),
    { 
      definition: 'randomPost: Post', 
      resolver: randomPostResolver
    }
  ],

  mutations: getDefaultMutations('Posts'),

});
```

**Comment by enzolutions** [comment]
This change should work with the current status of Vulcan, or will require changes in code source?

**Comment by SachaG** [comment]
No it'll be backwards-compatible so you won't have to do anything. I just think it's good to give the option to separate client/server code more cleanly if you need to.

**Comment by eric-burel** [comment]
Closing as this is included in 1.14 :tada:
